### PR TITLE
Make perlbrew bashrc compatible with bash -u

### PR DESCRIPTION
**Pull request by notpeter**
## **Problem statement**

1. Sourcing the provided `bashrc` from scripts where `set -u` (error on unset variables) will error.
2. Similarly, the bootstrap script also cannot be run with `curl -L https://install.perlbrew.pl | bash -eu` or similar.

## Solution

In bash scripts where unset variables are possible replace `"$VAR"` or `${VAR}` with `${VAR:-}`
This leverages bash default parameter substitution to explicitly provide a default value -- but because one is not provided it defaults to empty string (null).  In practice `${VAR:-}` is identical to the more explicit `${VAR:-''}` (empty string as default value).

Basically this makes explicit that having these variables unset/empty is safe as far as perlbrew is concerned:
```
PERLBREW_ROOT
PERLBREW_HOME
PERLBREW_PERL
PERLBREW_MANPATH
TMPDIR
```

## Result

This allows users to enable safer bash options (e.g. `set -euf -o pipefail`) in scripts dependent on perlbrew.

## References:

https://tldp.org/LDP/abs/html/parameter-substitution.html
https://sipb.mit.edu/doc/safe-shell/

### CODE REVIEW

**Comment by notpeter** [comment]
@gugod Let me know if there's anything I can do to make this merge-ready. Thanks!

**Comment by gugod** [comment]
@notpeter thank a lot on this work -- I haven't done much research myself and if there are something similar in fish / tcsh, we should porbably implement those too.

**Comment by gugod** [comment]
@notpeter I add a commit  b4577 which reapplies your changes in `perlbrew` to the soruce at `lib/App/perlbrew.pm`. The file `perlbrew` in a fatpack-ed version that is like the resulst out of compilation. If you need to add more patches, please only change the source under `lib/` . Thanks !
